### ISSUE TITLE
FR: Support for YCQL nested UDTs

### ISSUE THREAD

**Issue by balajivenki** [yugabyte/yugabyte-db#2729]
Hi,

Can you please support nested UDT support. 

E.g Table1 has frozen<udt1>, udt1 has frozen<udt2> and udt2 has list<frozen<udt3>>

**Comment by m-iancu** [comment]
@balajivenki thanks for reporting the issue, this is indeed a current limitation.

To clarify here is a concrete example:
- Create a simple UDT
    ```
    create keyspace test;
    use test;
    create type t1(a int, b int);
    ```
- Create another UDT that uses `t1`:
    ```
    create type t2(c int, d t1);
    ```
    will return:
    ```
    SyntaxException: Feature Not Supported. UDT field types cannot refer to other user-defined types
    create type t2(c int, d t1);
                          ^
     (ql error -14)
    ```
- Trying to use `frozen` for `t1` will not help:
    ```
    create type t2(c int, d frozen<t1>);
    ```
    ```
    SyntaxException: Feature Not Supported. UDT field types cannot refer to other user-defined types
    create type t2(c int, d frozen<t1>);
                          ^
     (ql error -14)
    ```

**Comment by m-iancu** [comment]
Fixed in 77a969a.
